### PR TITLE
[pigeon] Updates `PigeonInstanceMangerApi` to use the shared api channel code in Dart

### PR DESCRIPTION
**Pull request by bparrishMines**
This is in response to https://github.com/flutter/packages/pull/6371#discussion_r1605548447

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version
[following repository CHANGELOG style]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style
[exempt from CHANGELOG changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog
[test-exempt]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests

### CODE REVIEW

File: packages/pigeon/lib/dart_generator.dart [hunk 1]
```diff
@@ -859,7 +1051,7 @@ if (${varNamePrefix}replyList == null) {
       );
       indent.nest(2, () {
         final String channelSuffix =
-            addSuffixVariable ? '' : r'$messageChannelSuffix';
+            addSuffixVariable ? r'$messageChannelSuffix' : '';
```
[the remark this answers is not in the document]
Comment (tarrinneal) [reply packages/pigeon/lib/dart_generator.dart:1054]: how is it not breaking the tests?

[the remark this answers is not in the document]
Comment (bparrishMines) [reply packages/pigeon/lib/dart_generator.dart:1054]: I changed all the places where this variable was set to `true` to `false` and vice versa.

[the remark this answers is not in the document]
Comment (tarrinneal) [reply packages/pigeon/lib/dart_generator.dart:1054]: I must have changed the name and not the logic

**Review by tarrinneal — APPROVED** [review]
thanks for this

**Pull request merged**